src/thunderbots/software/util/visualizer_messenger/visualizer_messenger.h

```diff
@@ -0,0 +1,215 @@
+         * @param theta_start: The starting angle of the arc in rad (where 0 is pointed to
+         * +X horizontal heading, and PI/2 is south)
+         * @param theta_end: The ending angle of the arc in rad
+         * @param draw_style: the drawing style of the shape
+         * @param draw_transform: the transformation of the shape
+         */
+        void drawArc(const std::string& layer, double cx, double cy, double radius,
+                     double theta_start, double theta_end,
+                     DrawStyle draw_style         = DrawStyle(),
+                     DrawTransform draw_transform = DrawTransform());
+
+        /**
+         * Request a message to draw a line. The origin is the first point
+         *
+         * @param layer: The layer name this shape is being drawn to
+         * @param x1: Starting point X
+         * @param y1: Starting point Y
+         * @param x2: Ending point X
+         * @param y2: Ending point Y
+         * @param draw_style: the drawing style of the shape
+         * @param draw_transform: the transformation of the shape
+         */
+        void drawLine(const std::string& layer, double x1, double y1, double x2,
+                      double y2, DrawStyle draw_style = DrawStyle(),
+                      DrawTransform draw_transform = DrawTransform());
+
+       private:
+        /**
+         * Constructor; initializes an empty layers map then populates it
+         */
+        explicit VisualizerMessenger() : m_layers_name_to_msg_map(), m_publisher()
+        {
+            buildLayers();
+        }
+
+        /**
+         * Populates the layers map with some pre-defined layers
+         */
+        void buildLayers();
+
+        /**
+         * Helper function for the shape draw methods that copies the attributes in
+         * DrawStyle struct into the actual shape message
+         *
+         * @param shape_msg: Reference to the shape message
+         * @param style: The DrawStyle to be extracted
+         */
+        void applyDrawStyleToMsg(ShapeMsg& shape_msg, DrawStyle& style);
+
+        /**
+         * Helper function for the shape draw methods that copies the attributes in
+         * DrawTransform struct into the actual shape message
+         *
+         * @param shape_msg: Reference to the shape message
+         * @param transform: The DrawTransform struct to be extracted
+         */
+        void applyDrawTransformToMsg(ShapeMsg& shape_msg, DrawTransform& transform);
+
+        /**
+         * Helper function that checks if the layer exists,
+         * and push the shape message to the corresponding layer's shape vector.
+         *
+         * @param layer: The name of the layer to push to
+         * @param shape: The shape message
+         */
+        void addShapeToLayer(const std::string& layer, ShapeMsg& shape);
+
+       private:
+        // string to LayerMsg map
+        LayerMsgMap m_layers_name_to_msg_map;
+        ros::Publisher m_publisher;
```

Comment:
What does the `m_` prefix stand for? I'm guessing "member"? I don't think we use this convention anywhere else in the codebase, so for consistencies sake we should probably remove it if so?

Reply:
Shall I use `this->layers_name_to_msg_map` when accessing to make it clear that this is a member?

Reply:
I don't think we need to be _that_ explicit about what is a member variable. IDEs will highlight them differently, and it should be easy enough to tell that it's _not_ a local variable. In the rest of our code we just use them without a `m_` prefix or referencing with `this->`.

@garethellis0 thoughts?

Reply:
I'm not fussed either way. Adding `this->` is nice because it does make it more clear that it is a member variable, but it can impede readability a bit, and for me it's about an equal trade-off.